libvast/vast/transform.hpp

```diff
@@ -26,15 +39,23 @@ class transform {
 
   void add_step(transform_step_ptr step);
 
-  caf::expected<table_slice> apply(table_slice&&) const;
+  // FIXME: Convinence function
+  [[nodiscard]] caf::error add_slice(table_slice&&);
+  // FIXME: Convinence function
+  // The result is not ordered by offset
+  [[nodiscard]] caf::expected<std::vector<table_slice>> finish_slice();
 
   [[nodiscard]] const std::vector<std::string>& event_types() const;
 
   [[nodiscard]] const std::string& name() const;
 
 private:
-  [[nodiscard]] std::pair<vast::type, std::shared_ptr<arrow::RecordBatch>>
-  apply(vast::type layout, std::shared_ptr<arrow::RecordBatch> batch) const;
+  [[nodiscard]] caf::error add(vast::id offset, vast::type layout,
+                               std::shared_ptr<arrow::RecordBatch> batch);
+
+  [[nodiscard]] caf::expected<batch_vector> finish();
+
+  caf::error process_queue(const transform_step_ptr& step, batch_queue& queue);
```

Comment:
Please provide API docs.

From the docs it is not immediately clear to me who is owning the batch queue. It looks like the queue is an inout parameter. Should this instead be a member function of the class that actually owns the queue?

Reply:
The transform is operating on its own queue here, but yes, this should be `private`.

Reply:
The member variable `to_transform_` is the queue. So no need to pass it in as a parameter.